File: csharp/Library/LoanPeriod.cs
namespace SammanCoaching.Library;
public sealed record LoanPeriod
{
    public const int DefaultDays = 14;
    public const int MaxDays = 30;
    public LoanPeriod(int days)
    {
        if (days < 1)
        {
            throw new ArgumentException("Loan period must be at least one day", nameof(days));
        }

        if (days > MaxDays)
        {
            throw new ArgumentException($"Loan period must not be longer than {MaxDays} days", nameof(days));
        }

        Days = days;
    }
    public int Days { get; }
    public static LoanPeriod DefaultPeriod() => new(DefaultDays);
    public static LoanPeriod OfDays(int days) => new(days);
}

File: java/lib/src/main/java/org/sammancoaching/library/LoanPeriod.java
package org.sammancoaching.library;
public record LoanPeriod(int days) {
    public static final int DEFAULT_DAYS = 14;
    public static final int MAX_DAYS = 30;
    public LoanPeriod {
        if (days < 1) {
            throw new IllegalArgumentException("Loan period must be at least one day");
        }

        if (days > MAX_DAYS) {
            throw new IllegalArgumentException("Loan period must not be longer than " + MAX_DAYS + " days");
        }
    }
    public static LoanPeriod defaultPeriod() {
        return new LoanPeriod(DEFAULT_DAYS);
    }
    public static LoanPeriod ofDays(int days) {
        return new LoanPeriod(days);
    }
}
